_benchmark/reference/workloads/test-procedures.md

```diff
@@ -0,0 +1,193 @@
+:--- | :--- | :--- | :---
+`iterations` | No | Integer | The number of times a client should execute an operation. These are included in the measured results. Default is 1. 
+`warmup-iterations` | No | Integer | The number of times a client should execute an operation for the purpose of warming up the benchmark candidate. Warmup iterations will not show up in the measurement results. Default is 0.
+
+### Parallel tasks
+
+The `parallel` element runs tasks wrapped inside the element concurrently. 
+
+When running tasks in parallel, each task requires the `client` option to make sure clients inside your benchmark are reserved for that task. Otherwise, when the client option is specified inside the parallel element without a connection to the task, the benchmark will use that number of clients for all tasks.
+
+#### Usage
+
+In the following example, `parallel-task-1` and `parallel-task-2` execute a `bulk` operation concurrently:
+
+```yml
+{
+  "name": "parallel-any",
+  "description": "Workload completed-by property",
+  "schedule": [
+    {
+      "parallel": {
+        "tasks": [
+          {
+            "name": "parellel-task-1",
+            "operation": {
+              "operation-type": "bulk",
+              "bulk-size": 1000
+            },
+            "clients": 8
+          },
+          {
+            "name": "parellel-task-2",
+            "operation": {
+              "operation-type": "bulk",
+              "bulk-size": 500
+            },
+            "clients": 8
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+#### Options
+
+The `parallel` element supports all `schedule` parameters, in addition to the following:
+
+`tasks` | Yes | Array | Defines a list of tasks that should be executed concurrently. 
+`completed-by` | No | String | Allows you to define the name of one task in the tasks list, or the value `any`. If `completed-by` is set to the name of one of the tasks in the list, the parallel task structure is considered complete once that specific task has been completed. If `completed-by` is set to `any`, the parallel task structure is considered complete when any of the tasks in the list has been completed. If `completed-by` is not explicitly defined, the parallel task structure is considered complete as soon as all of the tasks in the list has been completed.
+
+### Time-based options
+
+Time-based options determines the duration of time, in seconds, that operations should run for. This is ideal for batch-style operations which may require an additional warmup period.
+
+To configure a time-based schedule, use the following options.
+
+Parameter | Required | Type | Description
+:--- | :--- | :--- | :---
+`time-period` | No | Integer | The time period in seconds that OpenSearch Benchmark considers for measurement. This is not required for bulk-indexing as OpenSearch Benchmark will bulk index all documents and naturally measure all samples after the `warmup-time-period` specified.
```

Comment:
```suggestion
`time-period` | No | Integer | Specifies the time period in seconds that OpenSearch Benchmark considers for measurement. This is not required for bulk indexing because OpenSearch Benchmark bulk indexes all documents and naturally measures all samples after the specified `warmup-time-period`.
```